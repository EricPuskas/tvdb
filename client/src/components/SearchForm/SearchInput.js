import React from "react";
import SearchBtn from "./SearchBtn";
import SearchSuggestions from "./SearchSuggestions";

const SearchInput = ({
  getInputProps,
  selectedItem,
  highlightedIndex,
  getItemProps,
  onSearchChange,
  searchText,
  isOpen,
  getListItemProps,
  items,
  loading
}) => {
  return (
    <>
      <div className="input-group mb-3">
        <input
          {...getInputProps({
            onChange: onSearchChange
          })}
          className="form-control"
          type="search"
          value={searchText}
          placeholder="Find your favorite shows"
          aria-label="Search Input"
        />
        <SearchBtn />
      </div>
      <SearchSuggestions
        loading={loading}
        searchText={searchText}
        getItemProps={getItemProps}
        selectedItem={selectedItem}
        highlightedIndex={highlightedIndex}
        isOpen={isOpen}
        getListItemProps={getListItemProps}
        items={items}
      />
    </>
  );
};

export default SearchInput;
