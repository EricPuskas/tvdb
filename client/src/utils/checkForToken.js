import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
