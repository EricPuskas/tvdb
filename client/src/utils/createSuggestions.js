const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
