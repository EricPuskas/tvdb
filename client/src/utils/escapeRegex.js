const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
