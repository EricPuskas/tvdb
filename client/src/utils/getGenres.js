import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
