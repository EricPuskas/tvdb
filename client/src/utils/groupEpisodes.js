const groupEpisodes = data => {
  let seasons = [];
  for (let i = 0; i < data.length; i++) {
    if (!seasons[data[i].airedSeason]) seasons[data[i].airedSeason] = [];
    seasons[data[i].airedSeason].push({
      id: data[i].id,
      season: data[i].airedSeason,
      episode: data[i].airedEpisodeNumber,
      title: data[i].episodeName
    });
  }
  return seasons;
};

export default groupEpisodes;
