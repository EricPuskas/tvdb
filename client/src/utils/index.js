import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
