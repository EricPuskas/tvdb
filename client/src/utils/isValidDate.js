import moment from "moment";

const isValidDate = date => {
  return moment(date).isValid();
};

export default isValidDate;
