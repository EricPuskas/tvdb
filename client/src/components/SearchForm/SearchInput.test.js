import React from "react";
import { shallow } from "enzyme";
import { findByAttribute } from "../../utils";
import SearchInput from "./SearchInput";

describe("SearchInput Component", () => {
  describe("Renders with props", () => {
    const props = {
      getInputProps: jest.fn(),
      selectedItem: "test",
      highlightedIndex: 0,
      getItemProps: jest.fn(),
      onSearchChange: jest.fn(),
      searchText: "String",
      isOpen: true,
      getListItemProps: jest.fn(),
      items: [1, 2, 3, 4],
      loading: false
    };
    const component = shallow(<SearchInput {...props} />);
    const wrapper = findByAttribute(component, "search-input");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });

    it("Should pass props from library to input", () => {
      expect(props.getInputProps).toHaveBeenCalled();
    });
  });
});
