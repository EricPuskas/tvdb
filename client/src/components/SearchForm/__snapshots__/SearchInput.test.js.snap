// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component Renders with props Should pass snapshot test 1`] = `"<div data-test=\\"search-input\\"><div class=\\"input-group mb-3\\"><input type=\\"search\\" class=\\"form-control\\" value=\\"String\\" placeholder=\\"Find your favorite shows\\" aria-label=\\"Search Input\\"/><div class=\\"input-group-append\\" data-test=\\"search-btn\\"><button class=\\"btn btn-primary\\" type=\\"submit\\" aria-label=\\"Search Button\\"><i class=\\"fas fa-search\\"></i> Search</button></div></div><div data-test=\\"search-suggestions\\"><div class=\\"Search-suggestions\\" style=\\"background-color:rgb(232, 232, 232);font-weight:normal\\">1</div><div class=\\"Search-suggestions\\" style=\\"background-color:#f9f9f9;font-weight:normal\\">2</div><div class=\\"Search-suggestions\\" style=\\"background-color:#f9f9f9;font-weight:normal\\">3</div><div class=\\"Search-suggestions\\" style=\\"background-color:#f9f9f9;font-weight:normal\\">4</div></div></div>"`;
